### PR TITLE
Use JavaScript Implementation of PayIDUtils

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Pull functionality from xpring-common-js rather than using a ILP library.

### Context of Change

We're going to change the PayID standard to be non-compliant with PaymentPointers and so the Java library no longer works.  

This is a pre-factor. A future PR will actually change the format in xpring-common-js and test cases. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

N/A

## Test Plan

CI

### CODE REVIEW

File: src/main/java/io/xpring/payid/PayIDComponents.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package io.xpring.payid;
+
+import org.immutables.value.Value;
+
+/**
+ * Represents components of a PayID.
+ */
+@SuppressWarnings("checkstyle:AbbreviationAsWordInName")
+@Value.Immutable
+public interface PayIDComponents {
```
Comment (sappenin) [inline src/main/java/io/xpring/payid/PayIDComponents.java:10]: You should just call this `PayId`

File: src/main/java/io/xpring/payid/PayIDClient.java [hunk 2]
```diff
@@ -71,7 +69,7 @@ public void setEnableSSLVerification(boolean enableSSLVerification) {
    * @return An XRP address representing the given PayID.
    */
   public String xrpAddressForPayID(String payID) throws PayIDException {
-    PaymentPointer paymentPointer = PayIDUtils.parsePayID(payID);
+    PayIDComponents paymentPointer = PayIDUtils.parsePayID(payID);
```
Comment (hbergren) [inline src/main/java/io/xpring/payid/PayIDClient.java:72]: Should we still be calling things `paymentPointer`s?

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java [hunk 3]
```diff
@@ -41,11 +39,11 @@ public void testParsePaymentPointerWithWellKnownPathAndTrailingSlash() {
     String rawPaymentPointer = "$example.com";
 
     // WHEN it is parsed to a PaymentPointer object
-    PaymentPointer paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
+    PayIDComponents paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
 
     // THEN the host and path are set correctly.
     assertEquals(paymentPointer.host(), "example.com");
-    assertEquals(paymentPointer.path(), PaymentPointer.WELL_KNOWN);
+    assertEquals(paymentPointer.path(), "/.well-known/pay");
```
Comment (hbergren) [inline src/test/java/io/xpring/payid/PayIDUtilsTest.java:46]: So the new payment pointer format comes later?

File: src/main/java/io/xpring/payid/javascript/JavaScriptPayIDUtils.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   * Parse the given Pay ID to a set of components.
+   *
+   * @param payID A PayID to parse.
+   * @return A set of components parsed from the PayID.
+   */
+  public PayIDComponents parsePayID(String payID) {
+    Objects.requireNonNull(payID);
+
+    Value parsePayIDFunction = javaScriptPayIDUtils.getMember("parsePaymentPointer");
+    Value javaScriptComponents =  parsePayIDFunction.execute(payID);
```
Comment (hbergren) [inline src/main/java/io/xpring/payid/javascript/JavaScriptPayIDUtils.java:45]: What's going on here?

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java [hunk 5]
```diff
@@ -15,7 +13,7 @@ public void testParsePaymentPointerHostAndPath() {
     String rawPaymentPointer = "$example.com/foo";
 
     // WHEN it is parsed to a PaymentPointer object
```
Comment (amiecorso) [inline src/test/java/io/xpring/payid/PayIDUtilsTest.java:15]: s/PaymentPoint/PayIDComponents

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java [hunk 6]
```diff
@@ -28,11 +26,11 @@ public void testParsePaymentPointerWithWellKnownPath() {
     String rawPaymentPointer = "$example.com";
 
     // WHEN it is parsed to a PaymentPointer object
```
Comment (amiecorso) [inline src/test/java/io/xpring/payid/PayIDUtilsTest.java:28]: also s/PaymentPointer/PayIDComponents

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java [hunk 7]
```diff
@@ -41,11 +39,11 @@ public void testParsePaymentPointerWithWellKnownPathAndTrailingSlash() {
...
```
Comment (amiecorso) [inline src/test/java/io/xpring/payid/PayIDUtilsTest.java:41]: same sub

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java [hunk 8]
```diff
@@ -54,7 +52,7 @@ public void testParsePaymentPointerIncorrectPrefix() {
     String rawPaymentPointer = "example.com/";
 
     // WHEN it is parsed to a PaymentPointer object
-    PaymentPointer paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
+    PayIDComponents paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
```
Comment (amiecorso) [inline src/test/java/io/xpring/payid/PayIDUtilsTest.java:55]: same sub

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java [hunk 9]
```diff
@@ -66,7 +64,7 @@ public void testParsePaymentPointerEmptyHost() {
     String rawPaymentPointer = "$";
 
     // WHEN it is parsed to a PaymentPointer object
-    PaymentPointer paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
+    PayIDComponents paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
```
Comment (amiecorso) [inline src/test/java/io/xpring/payid/PayIDUtilsTest.java:67]: etc

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java [hunk 10]
```diff
@@ -78,7 +76,7 @@ public void testParsePaymentPointerNonAscii() {
     String rawPaymentPointer = "$ZA̡͊͠͝LGΌ IS̯͈͕̹̘̱ͮ TO͇̹̺ͅƝ̴ȳ̳ TH̘Ë͖́̉ ͠P̯͍̭O̚N̐Y̡";
 
     // WHEN it is parsed to a PaymentPointer object
-    PaymentPointer paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
+    PayIDComponents paymentPointer = PayIDUtils.parsePayID(rawPaymentPointer);
```
Comment (amiecorso) [inline src/test/java/io/xpring/payid/PayIDUtilsTest.java:79]: one more

File: src/main/java/io/xpring/payid/PayIDComponents.java (hunk 1 above)
Comment (keefertaylor) [reply src/main/java/io/xpring/payid/PayIDComponents.java:10]: talked offline - using components for now.

File: src/test/java/io/xpring/payid/PayIDUtilsTest.java (hunk 3 above)
Comment (keefertaylor) [reply src/test/java/io/xpring/payid/PayIDUtilsTest.java:46]: Correct, #198

File: src/main/java/io/xpring/payid/PayIDClient.java (hunk 2 above)
Comment (keefertaylor) [reply src/main/java/io/xpring/payid/PayIDClient.java:72]: For now, spec is uptaken in #196

File: src/main/java/io/xpring/payid/javascript/JavaScriptPayIDUtils.java (hunk 4 above)
Comment (keefertaylor) [reply src/main/java/io/xpring/payid/javascript/JavaScriptPayIDUtils.java:45]: We're creating references to a JavaScript function and invoking it. In particular, the variable `javaScriptPayIDUtils` points to a JavaScript object of type `PayIDUtils` (in JS) and we retrieve a function called `parsePaymentPointer` which we store a reference to. 

Running `execute` passes the arguments to that function and gives us a pointer to the result.

**Pull request merged**